pkg/service/connect_service.go

```diff
@@ -0,0 +1,230 @@
+}
+
+func (s *ConnectService) ResolveBoolean(
+	ctx context.Context,
+	req *connect.Request[schemaV1.ResolveBooleanRequest],
+) (*connect.Response[schemaV1.ResolveBooleanResponse], error) {
+	res := connect.NewResponse(&schemaV1.ResolveBooleanResponse{})
+	result, variant, reason, err := s.Eval.ResolveBooleanValue(req.Msg.GetFlagKey(), req.Msg.GetContext())
+	if err != nil {
+		log.Error(err)
+		res.Msg.Reason = model.ErrorReason
+		return res, errFormat(err)
+	}
+	res.Msg.Reason = reason
+	res.Msg.Value = result
+	res.Msg.Variant = variant
+	return res, nil
+}
+
+func (s *ConnectService) ResolveString(
+	ctx context.Context,
+	req *connect.Request[schemaV1.ResolveStringRequest],
+) (*connect.Response[schemaV1.ResolveStringResponse], error) {
+	res := connect.NewResponse(&schemaV1.ResolveStringResponse{})
+	result, variant, reason, err := s.Eval.ResolveStringValue(req.Msg.GetFlagKey(), req.Msg.GetContext())
+	if err != nil {
+		log.Error(err)
+		res.Msg.Reason = model.ErrorReason
+		return res, errFormat(err)
+	}
+	res.Msg.Reason = reason
+	res.Msg.Value = result
+	res.Msg.Variant = variant
+	return res, nil
+}
+
+func (s *ConnectService) ResolveInt(
+	ctx context.Context,
+	req *connect.Request[schemaV1.ResolveIntRequest],
+) (*connect.Response[schemaV1.ResolveIntResponse], error) {
+	res := connect.NewResponse(&schemaV1.ResolveIntResponse{})
+	result, variant, reason, err := s.Eval.ResolveIntValue(req.Msg.GetFlagKey(), req.Msg.GetContext())
+	if err != nil {
+		log.Error(err)
+		res.Msg.Reason = model.ErrorReason
+		return res, errFormat(err)
+	}
+	res.Msg.Reason = reason
+	res.Msg.Value = result
+	res.Msg.Variant = variant
+	return res, nil
+}
+
+func (s *ConnectService) ResolveFloat(
+	ctx context.Context,
+	req *connect.Request[schemaV1.ResolveFloatRequest],
+) (*connect.Response[schemaV1.ResolveFloatResponse], error) {
+	res := connect.NewResponse(&schemaV1.ResolveFloatResponse{})
+	result, variant, reason, err := s.Eval.ResolveFloatValue(req.Msg.GetFlagKey(), req.Msg.GetContext())
+	if err != nil {
+		log.Error(err)
+		res.Msg.Reason = model.ErrorReason
+		return res, errFormat(err)
+	}
+	res.Msg.Reason = reason
+	res.Msg.Value = result
+	res.Msg.Variant = variant
+	return res, nil
+}
+
+func (s *ConnectService) ResolveObject(
+	ctx context.Context,
+	req *connect.Request[schemaV1.ResolveObjectRequest],
+) (*connect.Response[schemaV1.ResolveObjectResponse], error) {
+	res := connect.NewResponse(&schemaV1.ResolveObjectResponse{})
+	result, variant, reason, err := s.Eval.ResolveObjectValue(req.Msg.GetFlagKey(), req.Msg.GetContext())
+	if err != nil {
+		log.Error(err)
+		res.Msg.Reason = model.ErrorReason
+		return res, errFormat(err)
+	}
+	val, err := structpb.NewStruct(result)
+	if err != nil {
+		return res, err
+	}
+	res.Msg.Reason = reason
+	res.Msg.Value = val
+	res.Msg.Variant = variant
+	return res, nil
+}
+
+func newCORS() *cors.Cors {
+	// To let web developers play with the demo ConnectService from browsers, we need a
```

Comment:
Are there plans for this to be configurable?

Reply:
CORS will only tell browsers to ignore GET responses or prevent POST responses if the pre-flight fails Neither does it protect flagd from a load perspective, since as an attacker I can simply not use a browser and bypass CORS entirely that way.

CORS mitigates issues where malicious origin (webpage) X uses origin Y's credentials (stored in a cookie usually) to make a request to origin Y's backend, usually performing some kind of mutating action. That's not really applicable to us.

Reply:
Thinking about it more, you might be right @skyerus . I think there IS some possible DDoS protection here... If I infected a number of web pages (or host malicious web pages) that call a flagd I want to attack, I can use my origin/infected web page to tell  other users to spam that flagd target, adding load to it in a DDoS. Though I could attack flagd directly, my malicious origin is making my users attack that flagd too. CORS actually could help us with this, since a flagd would deny pre-flight requests from my hostile origins.

I think we should actually support CORS configuration, eventually, though it's probably something better configured in an LB or infra of some kind.

Reply:
I will make an issue to either support CORS, or document that it should be configured in infra.

Reply:
i initially added this config to allow for the flagd-web client demo to take place, happy to remove this until flagd-web is implemented, as im not sure how the external configuration for this should look as of yet

Reply:
That's interesting, I guess it would be useful in the scenario where a host exposes flagd directly to the public domain.

Reply:
This seem like something a reverse proxy should handle.

Reply:
+1 to CORS being handled at the reverse proxy...